odl/discr/discr_ops.py

```diff
@@ -8,88 +8,110 @@
 
 """Operators defined on `DiscreteLp`."""
 
-from __future__ import print_function, division, absolute_import
+from __future__ import absolute_import, division, print_function
+
 import numpy as np
 
-from odl.discr import DiscreteLp, uniform_partition
+from odl.discr.discr_utils import (
+    _all_interp_equal, _normalize_interp, per_axis_interpolator,
+    point_collocation)
+from odl.discr.lp_discr import DiscreteLp, uniform_partition
 from odl.operator import Operator
-from odl.set import IntervalProd
-from odl.space import FunctionSpace, tensor_space
+from odl.space import tensor_space
 from odl.util import (
-    normalized_scalar_param_list, safe_int_conv, writable_array, resize_array)
+    normalized_scalar_param_list, resize_array, safe_int_conv, writable_array)
 from odl.util.numerics import _SUPPORTED_RESIZE_PAD_MODES
-
+from odl.util.utility import none_context
```

Comment:
The whole thing is really an inferior backport of the Python 3.7+ built-in [`contextlib.nullcontext`](https://docs.python.org/3/library/contextlib.html#contextlib.nullcontext). I shied away from calling it the same because of a slightly wider API, but that should be very simple to implement.

Doing it "properly" would mean to have a `py_compat.py` module that would try to import the `contextlib` thing and fall back to a custom one if needed. We could also put the Py2 compatibility code there, e.g., `lru_cache`, `inspect.getfullargspec`, etc.

Does that sound better?

Reply:
OTOH, `lru_cache` is replaced with a dummy, and `getfullargspec` is used along with other Python version specific code, so they don't really fit into a `py_compat` module. Probably it's better to just put the code in `utility.py` for now.